docs/csharp/language-reference/keywords/readonly.md

```diff
@@ -14,7 +14,10 @@ ms.assetid: 2f8081f6-0de2-4903-898d-99696c48d2f4
 
 The `readonly` keyword is a modifier that can be used in three contexts:
 
-- In a [field declaration](#readonly-field-example), `readonly` indicates that assignment to the field can only occur as part of the declaration or in a constructor in the same class.
+- In a [field declaration](#readonly-field-example), `readonly` indicates that assignment to the field can only occur as part of the declaration or in a constructor in the same class. A readonly field can be assigned and reassigned multiple times within the field declaration and constructor. 
+  - If the `readonly` field is a value type, it is made immutable. 
+  - If the `readonly` field is a reference type, the field is not made immutable. The `readonly` prevents the field from being replaced by a different instance of the reference type. However, the modifier does not prevent the instance data of the field from being modified through the reference type. An externally visible type that contains an externally visible read-only field that is a mutable reference type may be a security vulnerability and may trigger warning [CA2104](../code-quality/ca2104-do-not-declare-read-only-mutable-reference-types.md) : "Do not declare read only mutable reference types."
```

Comment:
The previous comment applies to both paragraphs, but the GitHub UI doesn't allow that. Here's my suggestion for this bullet point:

```suggestion
- Because reference types contain a reference to their data, a field that is a `readonly` reference type must always refer to the same object. That object is not immutable. The `readonly` modifier prevents the field from being replaced by a different instance of the reference type. However, the modifier does not prevent the instance data of the field from being modified through the read-only field.

> [!WARNING]
> An externally visible type that contains an externally visible read-only field that is a mutable reference type may be a security vulnerability and may trigger warning [CA2104](../code-quality/ca2104-do-not-declare-read-only-mutable-reference-types.md) : "Do not declare read only mutable reference types."
```